tests/ubench/CMakeLists.txt

```diff
@@ -26,27 +26,8 @@ set(gbench_cmake_args
 
 
 # Attempt to update git submodule if required.
-find_package(Git)
-if(NOT EXISTS "${gbench_src_dir}/.git")
-    set(git_failed)
-    if(GIT_FOUND)
-        message(STATUS "Updating the google-benchmark submodule ${gbench_src_dir}")
-        execute_process(
-            COMMAND "${GIT_EXECUTABLE}" submodule update --init "${gbench_src_dir}"
-            WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
-            ERROR_VARIABLE git_error
-            RESULT_VARIABLE git_result)
-        if(NOT git_result EQUAL 0)
-            set(git_failed "${git_error}")
-        endif()
-    else()
-        set(git_failed "git not found")
-    endif()
-
-    if(git_failed)
-        message(WARNING "Unable to update the google-benchmark submodule: ${git_failed}")
-    endif()
-endif()
+include(GitSubmodule)
```

Comment:
Same with this... I can imagine situations where I don't want to update via git when I run CMake.

Reply:
This attempts to find git, but doesn't cause any errors if it doesn't find it.
How about we then make it only attempt to use git if the required dependency is not found *and* a target that needs a dependency is built, and then print a sane error message if it fails?